cli/src/commands/sign.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright 2023 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use clap_complete::ArgValueCandidates;
+use indexmap::IndexSet;
+use itertools::Itertools;
+use jj_lib::commit::Commit;
+use jj_lib::commit::CommitIteratorExt;
+use jj_lib::signing::SignBehavior;
+
+use crate::cli_util::CommandHelper;
+use crate::cli_util::RevisionArg;
+use crate::command_error::user_error_with_hint;
+use crate::command_error::CommandError;
+use crate::complete;
+use crate::ui::Ui;
+
+/// Cryptographically sign a revision
+///
+/// Note that revisions, which are already signed by you, will be resigned.
+///
+/// While that leads to discomfort for users, which sign with hardware devices,
+/// as of now we cannot reliably check if a commit is already signed by the user
+/// without creating a signature (see https://github.com/jj-vcs/jj/issues/5786).
```

Comment:
nit: maybe this can be added to the help of the `--revisions` argument? It's odd that the comment starts with `revisions` without specifying the details.

Reply:
Thanks, having it on the `--revisions` argument does put it into context.